File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int n = progresses.length;
        int[] daysToComplete = new int[n];

        for (int i = 0; i < n; i++) {
            int remainingProgress = 100 - progresses[i];
            daysToComplete[i] = (remainingProgress % speeds[i] == 0) ? (remainingProgress / speeds[i]) : (remainingProgress / speeds[i] + 1);
        }

        List<Integer> result = new ArrayList<>();
        int maxDays = daysToComplete[0];
        int count = 1;

        for (int i = 1; i < n; i++) {
            if (daysToComplete[i] <= maxDays) {
                count++;
            } else {
                result.add(count);
                maxDays = daysToComplete[i];
                count = 1;
            }
        }
        result.add(count);

        int[] resultArray = new int[result.size()];
        for (int i = 0; i < result.size(); i++) {
            resultArray[i] = result.get(i);
        }

        return resultArray;
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.kt
import java.util.Stack

class Solution {
    fun solution(progresses: IntArray, speeds: IntArray): IntArray {
        val daysToComplete = progresses.mapIndexed { index, progress ->
            val remainingProgress = 100 - progress
            if (remainingProgress % speeds[index] == 0) {
                remainingProgress / speeds[index]
            } else {
                remainingProgress / speeds[index] + 1
            }
        }

        val result = mutableListOf<Int>()
        var maxDays = daysToComplete[0]
        var count = 1

        for (i in 1 until daysToComplete.size) {
            if (daysToComplete[i] <= maxDays) {
                count++
            } else {
                result.add(count)
                maxDays = daysToComplete[i]
                count = 1
            }
        }
        result.add(count)

        return result.toIntArray()
    }
}